plugins/woocommerce/includes/class-wc-session-handler.php

```diff
@@ -496,24 +496,48 @@ private function is_session_expiring() {
 	 * For guests, sessions expire in 48 hours.
 	 */
 	public function set_session_expiration() {
-		$expiring_seconds   = DAY_IN_SECONDS;
-		$expiration_seconds = is_user_logged_in() ? WEEK_IN_SECONDS : 2 * DAY_IN_SECONDS;
-
 		/**
 		 * Filters the session expiration.
 		 *
 		 * @since 5.0.0
 		 * @param int $expiration_seconds The expiration time in seconds.
 		 */
-		$this->_session_expiring = time() + intval( apply_filters( 'wc_session_expiring', $expiring_seconds ) );
-
+		$expiring_seconds = intval( apply_filters( 'wc_session_expiring', DAY_IN_SECONDS ) );
```

Comment:
Should we make this more defensive in case the filter does not return an int?
What I'm thinking is to keep the default DAY_IN_SECONDS in a variable, and only overwrite if the filter returns an integer `intval( apply_filters( 'wc_session_expiring...`. Optionally we could check that the integer is not 0 to make sure that there was no error from intval?